src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,107 @@
+/*
+    Jon Kaimmer
+    CS410 Software Engineering
+    Spring 2023
+    Junit Testing Assignment
+
+    Spring Manipulation
+ */
+
+import java.lang.IllegalArgumentException;
+import java.lang.IndexOutOfBoundsException;
+import java.lang.NullPointerException;
+import java.util.*;
+
+
 public class StringManipulation implements StringManipulationInterface {
 
+    public String s;
+
     @Override
     public String getString() {
-        return null;
+        return s;
     }
 
     @Override
     public void setString(String string) {
+        s = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (s == null || s.isEmpty()){
+            return 0;
+        }
+        String[] words = s.split("\\s+");
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+
+        if(s == null){
+            throw new NullPointerException("string cannot be null");
```

Comment:
Null ptr exception not mentioned in the interface for the assigment.
Throw indexoutofbound? null str has no words

Reply:
Yeah you are probably right, I just wanted to add exception checking and figured that a string might be null so why not check whether the string was nulll.. indexOutOfBounds is a good suggestion